nemo/collections/llm/gpt/model/hyena.py

```diff
@@ -0,0 +1,698 @@
+                        f'decoder.layers.{i}.mixer.mixer.short_conv.short_conv_weight'
+                    )
+
+                elif symbol == 'D':
+                    mapping[f'sequential.{i}.mixer.mixer.conv_bias'] = f'decoder.layers.{i}.mixer.mixer.conv_bias'
+                    mapping[f'sequential.{i}.mixer.mixer.filter.h'] = f'decoder.layers.{i}.mixer.mixer.filter.h'
+                    mapping[f'sequential.{i}.mixer.mixer.filter.decay'] = (
+                        f'decoder.layers.{i}.mixer.mixer.filter.decay'
+                    )
+
+                elif symbol == 'H':
+                    mapping[f'sequential.{i}.mixer.mixer.conv_bias'] = f'decoder.layers.{i}.mixer.mixer.conv_bias'
+                    mapping[f'sequential.{i}.mixer.mixer.filter.gamma'] = (
+                        f'decoder.layers.{i}.mixer.mixer.filter.gamma'
+                    )
+                    mapping[f'sequential.{i}.mixer.mixer.filter.R'] = f'decoder.layers.{i}.mixer.mixer.filter.R'
+                    mapping[f'sequential.{i}.mixer.mixer.filter.p'] = f'decoder.layers.{i}.mixer.mixer.filter.p'
+
+            elif symbol == '*':
+                if te_enabled:
+                    mapping[f'sequential.{i}.input_layernorm.weight'] = (
+                        f'decoder.layers.{i}.self_attention.linear_qkv.layer_norm_weight'
+                    )
+                else:
+                    mapping[f'sequential.{i}.input_layernorm.weight'] = f'decoder.layers.{i}.input_layernorm.weight'
+
+                mapping[f'sequential.{i}.mixer.dense_projection.weight'] = (
+                    f'decoder.layers.{i}.self_attention.linear_qkv.weight'
+                )
+                mapping[f'sequential.{i}.mixer.dense.weight'] = f'decoder.layers.{i}.self_attention.linear_proj.weight'
+                mapping[f'sequential.{i}.mixer.dense.bias'] = f'decoder.layers.{i}.self_attention.linear_proj.bias'
+            else:
+                raise ValueError(f'Unknown symbol: {symbol}')
+
+        return io.apply_transforms(source, target, mapping=mapping, transforms=[_import_linear_fc1])
+
+    @property
+    def tokenizer(self):
+        """
+        Gets the tokenizer for the model.
+
+        Returns:
+            Tokenizer instance
+        """
+        from nemo.collections.nlp.modules.common.tokenizer_utils import get_nmt_tokenizer
+
+        tokenizer = get_nmt_tokenizer(
+            library=self.model_config.tokenizer_library,
+        )
+
+        return tokenizer
+
+    @property
+    def config(self) -> HyenaConfig:
+        """
+        Gets the model configuration.
+
+        Returns:
+            HyenaConfig: Model configuration
+        """
+        return self.model_config
+
+
+@io.state_transform(
+    source_key=("sequential.*.mlp.w1.weight", "sequential.*.mlp.w2.weight"),
+    target_key="decoder.layers.*.mlp.linear_fc1.weight",
+)
+def _import_linear_fc1(w1, w2):
```

Comment:
can you switch to using TransformFns.merge_fc1?

Reply:
Good idea. Thanks!